builtin/mainmenu/tab_settings.lua

```diff
@@ -15,497 +15,446 @@
 --with this program; if not, write to the Free Software Foundation, Inc.,
 --51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
---------------------------------------------------------------------------------
-
-local leaves_style_labels = {
-	fgettext("Opaque Leaves"),
-	fgettext("Simple Leaves"),
-	fgettext("Fancy Leaves")
-}
-
-local leaves_style = {
-	{leaves_style_labels[1]..","..leaves_style_labels[2]..","..leaves_style_labels[3]},
-	{"opaque", "simple", "fancy"},
-}
-
-local dd_filter_labels = {
-	fgettext("No Filter"),
-	fgettext("Bilinear Filter"),
-	fgettext("Trilinear Filter")
-}
-
-local filters = {
-	{dd_filter_labels[1]..","..dd_filter_labels[2]..","..dd_filter_labels[3]},
-	{"", "bilinear_filter", "trilinear_filter"},
-}
-
-local dd_mipmap_labels = {
-	fgettext("No Mipmap"),
-	fgettext("Mipmap"),
-	fgettext("Mipmap + Aniso. Filter")
-}
-
-local mipmap = {
-	{dd_mipmap_labels[1]..","..dd_mipmap_labels[2]..","..dd_mipmap_labels[3]},
-	{"", "mip_map", "anisotropic_filter"},
-}
-
-local function getLeavesStyleSettingIndex()
-	local style = core.setting_get("leaves_style")
-	if (style == leaves_style[2][3]) then
-		return 3
-	elseif (style == leaves_style[2][2]) then
-		return 2
+local function parseConfigFile()
+	local file = io.open(core.get_builtin_path() .. DIR_DELIM .. "settings.txt", "r")
+	local settings = {}
+	if not file then
+		core.log("error", "Can't load settings.txt")
+		return settings
 	end
-	return 1
-end
 
-local dd_antialiasing_labels = {
-	fgettext("None"),
-	fgettext("2x"),
-	fgettext("4x"),
-	fgettext("8x"),
-}
+	local currentComment = ""
+	
+	local line = file:read("*line")
+	while line do
```

Comment:
Perhaps the body of the while can be put into its own function? Then you wouldn't need the `if handled then`, and could simply do `return true` instead of `handled = true`.